clevertap-core/src/main/java/com/clevertap/android/sdk/CoreState.java

```diff
@@ -313,6 +316,14 @@ public void setCryptHandler(final CryptHandler cryptHandler) {
         this.cryptHandler = cryptHandler;
     }
 
+    public TemplatesManager getTemplatesManager() {
+        return templatesManager;
+    }
+
+    public void setTemplatesManager(final TemplatesManager templatesManager) {
```

Comment:
is this setter needed or we always pass it in constructor?

Reply:
The constructor of CoreState is only passed a Context. All other fields are set in CleverTapFactory::getCoreState with setters, the same is done with the TemplatesManager.